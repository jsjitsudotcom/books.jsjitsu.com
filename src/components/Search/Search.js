import React, { PureComponent } from "react";
import Style from "./Search.scss";
import PropTypes from "prop-types";

import searchIcon from "./../../assets/material-icons/search.svg";

export default class Search extends PureComponent {
  render() {
    return (
      <div className={Style.container}>
        <label className={Style.icon} htmlFor="search">
          <img src={searchIcon} alt="search" />
        </label>
        <input
          id="search"
          placeholder="Quel livre recherchez vous ?"
          value={this.props.value}
          className={Style.input}
          onChange={this.props.onChange}
        />
      </div>
    );
  }
}

Search.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func
};

Search.defaultProps = {
  value: "",
  onChange: /* istanbul ignore next */ () => false
};
