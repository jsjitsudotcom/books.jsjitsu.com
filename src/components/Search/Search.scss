:local(.container) {
  width: 100%;
  display: flex;
  align-items: center;
}

:local(.icon) {
  display: flex;
  align-items: center;

  img {
    display: block;
  }
}

:local(.input) {
  background-color: transparent;
  border: none;
  font-size: 16px;
  padding-left: 10px;
  width: 100%;
}
