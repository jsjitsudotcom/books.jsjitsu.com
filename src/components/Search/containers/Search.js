import { connect } from "react-redux";
import actions from "./../../../actions";

const mapState = (state, props) => ({
  value: state.books.search
});

const mapDispatch = (dispatch, props) => ({
  onChange: value => dispatch(actions.books.changeSearchAndFetch(value))
});

export default connect(mapState, mapDispatch);
