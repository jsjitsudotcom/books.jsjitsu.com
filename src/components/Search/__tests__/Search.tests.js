import React from "react";
import { shallow } from "enzyme";
import sinon from "sinon";
import Search from "./../Search";

describe("<Search />", () => {
  it("Le componsant doit être rendu", () => {
    shallow(<Search />);
  });

  describe("Suite de tests de l'affichage", () => {
    test("Le composant doit bien s'afficher", () => {
      const value = "value";

      const wrapper = shallow(
        <Search value={value} />
      );

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("Suite test des methodes", () => {
    test("La propriété onChange doit être appelée", () => {
      const onChange = sinon.spy();

      const wrapper = shallow(<Search onChange={onChange} />);

      wrapper.find(".input").prop("onChange")();

      expect(onChange.calledOnce).toEqual(true);
    });
  });
});
