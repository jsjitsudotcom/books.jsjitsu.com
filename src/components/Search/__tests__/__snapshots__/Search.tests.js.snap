// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> Suite de tests de l'affichage Le composant doit bien s'afficher 1`] = `
<div
  className="container"
>
  <label
    className="icon"
    htmlFor="search"
  >
    <img
      alt="search"
      src="search.svg"
    />
  </label>
  <input
    className="input"
    id="search"
    onChange={[Function]}
    placeholder="Quel livre recherchez vous ?"
    value="value"
  />
</div>
`;
